src/libstore/tests/derivation.cc

```diff
@@ -89,28 +83,20 @@ TEST_F(DynDerivationTest, BadATerm_oldVersionDynDeps) {
                 mockXpSettings);                                          \
             DerivationOutput expected { VAL };                            \
             ASSERT_EQ(got, expected);                                     \
-        }                                                                 \
+        });                                                               \
     }                                                                     \
                                                                           \
     TEST_F(FIXTURE, DerivationOutput_ ## NAME ## _to_json) {              \
-        auto file = goldenMaster("output-" #NAME ".json");                \
-                                                                          \
-        json got = DerivationOutput { VAL }.toJSON(                       \
-            *store,                                                       \
-            DRV_NAME,                                                     \
-            OUTPUT_NAME);                                                 \
-                                                                          \
-        if (testAccept())                                                 \
-        {                                                                 \
-            createDirs(dirOf(file));                                      \
-            writeFile(file, got.dump(2) + "\n");                          \
-            GTEST_SKIP() << updatingGoldenMaster;                         \
-        }                                                                 \
-        else                                                              \
-        {                                                                 \
-            auto expected = json::parse(readFile(file));                  \
-            ASSERT_EQ(got, expected);                                     \
-        }                                                                 \
+        writeTest<json>("output-" #NAME ".json", [&]() -> json {          \
+            return DerivationOutput { VAL }.toJSON(                       \
```

Comment:
```suggestion
            return DerivationOutput { (VAL) }.toJSON(                       \
```

The parentheses prevent `VAL` to be interpreted as multiple parameters if someone accidentally found a way to enter something with commas